File: auth-service/src/main/java/com/pouetteur/authservice/model/Community.java
package com.pouetteur.authservice.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "communities")
public class Community implements Profile{
    @Id
    @GeneratedValue(generator = "IdGenerator")
    @GenericGenerator(name = "IdGenerator", strategy = "com.pouetteur.authservice.config.IdGenerator")
    private String id;
    private String username;
    private String name;
    private Date creationDate;
    private String profilePhoto;
    private String profileBanner;
    private String bio;
    @ManyToMany
    private ArrayList<Member> members;
    @ManyToOne
    private Member owner;

    @Override
    protected Object clone() throws CloneNotSupportedException {
        return super.clone();
    }
}

File: profile-service/profile-service/Models/Community.cs
using profile_service.DTOs;

namespace profile_service.Models;

public class Community : IProfile
{
    public string? Id { get; set; }
    public string? Username { get; set; }
    public string? Name { get; set; }
    public DateTime? CreationDate { get; set; }
    public string? ProfilePhoto { get; set; }
    public string? ProfileBanner { get; set; }
    public string? Bio { get; set; }
    public List<Member>? Members { get; set; }
    public Member? Owner { get; set; }
    
    public Community(string username, string name, DateTime creationDate, Member owner)
    {
        Id = "m*" + Guid.NewGuid();
        Username = username;
        Name = name;
        CreationDate = creationDate;
        Owner = owner;
        Members = new List<Member>();
    }

    public Community(ProfileDto profileDto)
    {
        Id = "m*" + Guid.NewGuid();
        Username = profileDto.Username;
        Name = profileDto.Name;
        CreationDate = DateTime.Now;
        ProfilePhoto = profileDto.ProfilePhoto;
        ProfileBanner = profileDto.ProfileBanner;
        Bio = Bio == null ? "" : profileDto.Bio;
        Members = new List<Member>();
    }

    public Community(){}

}